src/components/CodeView/index.spec.tsx

```diff
@@ -333,6 +334,68 @@ describe(__filename, () => {
     expect(message.at(1).prop('message')).toMatchObject({ uid: secondUid });
   });
 
+  it('renders a scrollTo div at the top of global LinterMessage components, if line 0 is selected', () => {
+    const firstUid = 'first-uid';
+    const secondUid = 'second-uid';
+    const id = getCodeLineAnchorID(0);
+    const location = createFakeLocation({ hash: `#${id}` });
+
+    const { root } = renderWithMessages({
+      location,
+      messages: [
+        createGlobalExternalMessage({ uid: firstUid }),
+        createGlobalExternalMessage({ uid: secondUid }),
+      ],
+    });
+
+    expect(root.find(`#${id}`)).toHaveLength(1);
+  });
+
+  it('does not render a scrollTo div at the top of global LinterMessage components, if line 0 is not selected', () => {
+    const uid = 'some-global-message-id';
+    const externalMessage = createGlobalExternalMessage({ uid });
+    const location = createFakeLocation({ hash: '#I1' });
+
+    const { root } = renderWithMessages({
+      location,
+      messages: [externalMessage],
+    });
+
+    expect(root.find(`#${getCodeLineAnchorID(0)}`)).toHaveLength(0);
+  });
+
+  it('does not render a scrollTo div at the top of global LinterMessage components, if there are no global messages', () => {
+    const id = getCodeLineAnchorID(0);
+    const location = createFakeLocation({ hash: `#${id}` });
+
+    const { root } = renderWithMessages({
+      location,
+      messages: [{ line: 1 }],
+    });
+
+    expect(root.find(`#${id}`)).toHaveLength(0);
+  });
+
+  it('calls _scrollToSelectedLine() when rendering a selected global message', () => {
+    const uid = 'some-global-message-id';
+    const externalMessage = createGlobalExternalMessage({ uid });
+    const location = createFakeLocation({ hash: `#${getCodeLineAnchorID(0)}` });
+    const _scrollToSelectedLine = jest.fn();
+
+    const { root } = renderWithMessages({
+      _scrollToSelectedLine,
+      location,
+      messages: [externalMessage],
+    });
+
+    // TODO: This doesn't work because we need `mount()` as `ref` is only used in a DOM environment.
```

Comment:
Did you try the mock ref approach? https://github.com/mozilla/addons-code-manager/blob/76866e1634f73e5b3fc9629ef9095c81b77c9bdb/src/components/FileTreeNode/index.spec.tsx#L719

Reply:
I want to make sure I understand what's being suggested. From what I can tell, if I want to use the mock ref approach I also need to adopt a bunch of code similar to what is in `FileTreeNode`, correct? I.e., I'm going to need to add code for `componentDidMount` and `componentDidUpdate` which calls a function which evaluates whether I need to scroll the node in question into view?

What we're doing currently, which is simply assigning `_scrollToSelectedLine` to the `ref` property of the element if we know we want to scroll to it - that will no longer be possible if we want to be able to inject a mock ref. Is that accurate? I just want to check before I go and make a bunch of changes.

I assume that if I am making this change I should apply it not just to the `ref` for the `div` for global messages, but also to the `ref` which is potentially added to a table row at https://github.com/mozilla/addons-code-manager/blob/76866e1634f73e5b3fc9629ef9095c81b77c9bdb/src/components/CodeView/index.tsx#L95. Do you agree?

Reply:
Ah, I understand the problem better now.

Ideally, we would use enzyme to shallow render and check that the `ref` prop was specified only for the selected state. This not supported by enzyme and there are lots of closed issues about it.

On second thought, I don't think switching to a mock ref object is worth the effort. You are right that it would require some redesign -- the addition of new mount/update logic.

The second best thing is to mount the component to check that the `ref` gets set only in the selected state. On my previous look, I missed that code [for mounting](https://github.com/mozilla/addons-code-manager/blob/365b8958bd517c444f4af1008a0eb3beb211cb4b/src/components/CodeView/index.spec.tsx#L84) already exists! Can't you just re-use that function but dispatch a linter result right before calling it? Like this:

https://github.com/mozilla/addons-code-manager/blob/365b8958bd517c444f4af1008a0eb3beb211cb4b/src/components/LinterProvider/index.spec.tsx#L107

Reply:
Yes, I was able to get that to work, thanks!